test-data/unit/check-python38.test

```diff
@@ -198,6 +198,27 @@ if a := 2:
 while b := "x":
     reveal_type(b)  # N: Revealed type is 'builtins.str'
 
+l = [y2 := 1, y2 + 2, y2 + 3]
+reveal_type(y2)  # N: Revealed type is 'builtins.int'
+reveal_type(l)  # N: Revealed type is 'builtins.list[builtins.int*]'
+ 
+filtered_data = [y3 for x in l if (y3 := a) is not None]
+reveal_type(filtered_data)  # N: Revealed type is 'builtins.list[builtins.int*]'
+reveal_type(y3)  # N: Revealed type is 'builtins.int'
+
+d = {'a': (a2 := 1), 'b': a2 + 1, 'c': a2 + 2}
+reveal_type(d)  # N: Revealed type is 'builtins.dict[builtins.str*, builtins.int*]'
+reveal_type(a2)  # N: Revealed type is 'builtins.int'
+
+d2 = {(prefix := 'key_') + 'a': (start_val := 1), prefix + 'b': start_val + 1, prefix + 'c': start_val + 2}
+reveal_type(d2)  # N: Revealed type is 'builtins.dict[builtins.str*, builtins.int*]'
+reveal_type(prefix)  # N: Revealed type is 'builtins.str'
+reveal_type(start_val)  # N: Revealed type is 'builtins.int'
+
+filtered_dict = {k: new_v for k, v in [('a', 1), ('b', 2), ('c', 3)] if (new_v := v + 1) == 2}
```

Comment:
If reviewers are wondering why I didn't just use `d.items()` here instead of the list of tuples, it's because it's causing `error: "Dict[str, int]" has no attribute "items"`.
I couldn't figure out why that was happening, I tried different flags and casting and whatnot but something in the test environment must be different... in a regular python script `filtered_dict = {k: new_v for k, v in d.items() if (new_v := v + 1) == 2}` can have its type revealed without such an error. Open to suggestions if you have ideas what's causing that!

Reply:
This is almost certainly because the test fixture you're using happens not to define `dict.items`. What you're doing here is fine.